### ISSUE TITLE
Aid regression testing to identify performance differences at op and module granularity

### ISSUE THREAD

**Issue by ppnaik1890** [torch-spyre/torch-spyre#1591]
Add a performance regression test harness that captures per-op and per-module
duration baselines and fails when a subsequent run exceeds a configurable
threshold.

**Proposed design**

1. **Baseline capture** — a script or pytest fixture that runs a set of
   benchmark models/ops, collects `KernelMetrics` (or `ENABLE_TRACING=2`
   log output), and writes a JSON baseline file:

   ```json
   {
     "torch.matmul_32x32": {"duration_ms_p50": 1.2, "duration_ms_p95": 1.5},
     "MyModel.forward": {"duration_ms_p50": 45.0, "duration_ms_p95": 52.0}
   }
   ```

2. **Regression check** — a pytest parametrized test reads the baseline and
   asserts that each entry's measured duration is within `threshold` (default
   10 %) of baseline. Threshold is configurable via a marker or env var
   (`PERF_REGRESSION_THRESHOLD=0.15`).

3. **Granularity levels**:
   - *Op level*: individual ATen ops (matmul, add, layernorm, …)
   - *Module level*: full `nn.Module` forward passes (MLP, attention block, …)

4. **CI integration**: the regression test suite runs in a separate CI job
   gated on hardware availability (`@pytest.mark.requires_spyre`). Baseline
   files are committed to `tests/perf_baselines/`.

**Acceptance criteria**
- [ ] Baseline capture script produces a valid JSON file for a representative
  set of ops (at minimum: matmul, add, layernorm, softmax) and one module.
- [ ] Regression test fails with a clear message when a measured duration
  exceeds the baseline threshold.
- [ ] Regression test passes when durations are within threshold.
- [ ] Threshold is configurable without code changes.
- [ ] Baselines are stored in version control at `tests/perf_baselines/`.
- [ ] Documentation updated to describe how to capture a new baseline and
  how to update baselines intentionally.

**Comment by ppnaik1890** [comment]
Closing since handled in perf suite
